lua/presence/init.lua

```diff
@@ -613,8 +615,65 @@ function Presence.discord_event(on_ready)
     end
 end
 
+-- Check if the current project/parent is in blacklist
+function Presence:check_blacklist(buffer, parent_dirpath, project_dirpath)
+    local parent_dirname = nil
+    local project_dirname = nil
+
+    -- Parse parent/project directory name
+    if parent_dirpath then
+        parent_dirname = self.get_filename(parent_dirpath, self.os.path_separator)
+    end
+
+    if project_dirpath then
+        project_dirname = self.get_filename(project_dirpath, self.os.path_separator)
+    end
+
+    -- Blacklist table
+    local blacklist_table = self.options["blacklist"]
+
+    -- Loop over the values to see if the provided project/path is in the blacklist
+    for _, val in pairs(blacklist_table) do
+        -- Match buffer
+        if buffer:match(val) == buffer then return true end
+        -- Match parent
+        if parent_dirpath then
+            if parent_dirpath:match(val) == parent_dirpath or parent_dirname:match(val) == parent_dirname then return true
+            end
+        end
+        -- Match project
+        if project_dirpath then
+            if project_dirpath:match(val) == project_dirpath or project_dirname:match(val) == project_dirname then return true
+            end
+        end
+    end
+
+    return false
+end
+
 -- Update Rich Presence for the provided vim buffer
 function Presence:update_for_buffer(buffer, should_debounce)
+    -- Parse vim buffer
+    local filename = self.get_filename(buffer, self.os.path_separator)
+    local parent_dirpath = self.get_dir_path(buffer, self.os.path_separator)
+    local extension = filename and self.get_file_extension(filename) or nil
+
+    self.log:debug(string.format("Parsed filename %s with %s extension", filename, extension or "no"))
+
+    self.log:debug(string.format("Getting project name for %s...", parent_dirpath))
+
+    local project_name, project_path = self:get_project_name(parent_dirpath)
+
+    -- Check for blacklist
+    local is_blacklisted = self:check_blacklist(buffer, parent_dirpath, project_path)
```

Comment:
It would probably make sense to only check the blacklist only if there are actually items in the blacklist, so we could just include a condition here:
```suggestion
    local is_blacklisted = #self.options.blacklist > 0 and self:check_blacklist(buffer, parent_dirpath, project_path)
```